### ISSUE TITLE
Very high CPU-usage during webcam playback

### ISSUE THREAD

**Issue by UnitedMarsupials** [mpv-player/mpv#6081]
### mpv version and platform

0.29

### Reproduction steps

Trying to play the video-capture from my "webcam":

`mpv tv://`

It works, and the picture is quite good. But the CPU-usage is at 100% (on a Xeon E5-1620 0 @ 3.60GHz).

ffplay and mplayer can both play the same stream using under 10% of the CPU -- however, their output is quite hideous. mpv must be doing something smart.

Trying different video-outputs:

- gpu (default)
- xv
- vdpau
- vaapi

does not seem to make a difference.

### Expected behavior

CPU-usage of 1-10%

### Actual behavior

CPU-usage at 100%

### Log file

[mpv.log](https://github.com/mpv-player/mpv/files/2303430/mpv.log)

### Sample files

N/A

### Operating system

FreeBSD-11.x stable, amd64

### Video card

ATI Radeon HD 5670, driven by xf86-video-ati-7.9.0

**Comment by UnitedMarsupials** [comment]
Actually, now that you ask, it is reproducible with just about _any_ video-file I give it...

**Comment by qmega** [comment]
Can you see the name of the thread that's using all the cpu?

If you pause the video, does the usage go down or does it stay at 100?

**Comment by UnitedMarsupials** [comment]
While playing, there many threads named simply "{mpv}" in top. The one named "{radeon_cs:0}" is not very active at all, consuming only %0.22 of the CPU.

If I pause the video, the CPU-usage drops down to zero, the mpv-process is in the state `uwait`.

**Comment by qmega** [comment]
(Forgot about this, sorry. Though honestly I'm not the best person to help with this anyway.)

Just to make sure, are you getting any dropped frames or a/v desync? For example, with the end of the status line `A-V: -0.009 DS: 3.000/4 Dropped: 1` the -0.009 is desync, and the 4 and 1 are dropped frames. If any of those numbers are high or continuously rising for you, your machine isn't actually capable of playing the video in real time and so max CPU usage would be understandable.

> many threads named simply "{mpv}" in top

Hmm... They should have individual names like `mpv/vo`. Maybe FreeBSD's `top` doesn't show pthread names? You could try htop maybe? (requires some Linux compat stuff but there's a port)

A VO thread seems most likely, though. I would guess at a busy wait on vsync. That could probably be blamed on the graphics driver, especially since it happens with any VO you try, but I don't know much about that part of mpv's code so don't take my word for it.

Could you attach a debugger to it and backtrace it (all threads) to see where it's stuck? You'll want to try a couple times since you could end up in actual drawing code by chance.

**Comment by Akemi** [comment]
if this issue still persists, please open a new issue.
